Avoids crash when F12 is pressed during installation of packages.

From c5886c50e5c4498d8f60c2facddc9c3fe52282d6 Mon Sep 17 00:00:00 2001
From: David Shea <dshea@redhat.com>
Date: Fri, 21 Aug 2015 13:54:10 -0400
Subject: [PATCH] Start hubs with the buttons insensitive.

Just because you can't see the buttons doesn't mean you can't click them
(because we attached keyboard shortcuts to them). With the progress hub,
this meant that hitting F12 would reboot in the middle of installing
packages, and in the case of the summary hub there was a potential race
between the hub being created and the button states being initialized.

Start the hubs with the all of the buttons insensitive, and let the code
that was mostly already there update the sensitivities as the status of
the hub changes.

(cherry picked from commit e92ecaf261f9fca13ee153128010d70c6e5172e2)
---
 pyanaconda/ui/gui/hubs/progress.glade | 2 ++
 pyanaconda/ui/gui/hubs/progress.py    | 1 +
 pyanaconda/ui/gui/hubs/summary.glade  | 6 ++----
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/pyanaconda/ui/gui/hubs/progress.glade b/pyanaconda/ui/gui/hubs/progress.glade
index cabd07f74..39d743166 100644
--- a/pyanaconda/ui/gui/hubs/progress.glade
+++ b/pyanaconda/ui/gui/hubs/progress.glade
@@ -98,6 +98,7 @@ Finish it and then click the Finish configuration button please.</property>
                                 <property name="label" translatable="yes" context="GUI|Progress">_Finish configuration</property>
                                 <property name="use_action_appearance">False</property>
                                 <property name="visible">True</property>
+                                <property name="sensitive">False</property>
                                 <property name="can_focus">True</property>
                                 <property name="receives_default">True</property>
                                 <property name="halign">end</property>
@@ -151,6 +152,7 @@ Go ahead and reboot to start using it!</property>
                                 <property name="label" translatable="yes" context="GUI|Progress">_Reboot</property>
                                 <property name="use_action_appearance">False</property>
                                 <property name="visible">True</property>
+                                <property name="sensitive">False</property>
                                 <property name="can_focus">True</property>
                                 <property name="receives_default">True</property>
                                 <property name="halign">end</property>
diff --git a/pyanaconda/ui/gui/hubs/progress.py b/pyanaconda/ui/gui/hubs/progress.py
index 3f915137c..e21c0e827 100644
--- a/pyanaconda/ui/gui/hubs/progress.py
+++ b/pyanaconda/ui/gui/hubs/progress.py
@@ -132,6 +132,7 @@ class ProgressHub(Hub):
 
         GLib.source_remove(self._rnotes_id)
         self._progressNotebook.set_current_page(1)
+        self.window.set_may_continue(True)
 
         iutil.ipmi_report(IPMI_FINISHED)
 
diff --git a/pyanaconda/ui/gui/hubs/summary.glade b/pyanaconda/ui/gui/hubs/summary.glade
index 88313c248..447d4eec5 100644
--- a/pyanaconda/ui/gui/hubs/summary.glade
+++ b/pyanaconda/ui/gui/hubs/summary.glade
@@ -24,9 +24,6 @@
                 <child>
                   <placeholder/>
                 </child>
-                <child>
-                  <placeholder/>
-                </child>
               </object>
             </child>
           </object>
@@ -100,6 +97,7 @@
                 <property name="label" translatable="yes" context="GUI|Summary">_Begin Installation</property>
                 <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
+                <property name="sensitive">False</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">True</property>
                 <property name="use_underline">True</property>
@@ -127,8 +125,8 @@
             <property name="can_focus">False</property>
             <property name="margin_right">6</property>
             <property name="margin_bottom">6</property>
-            <property name="xalign">1</property>
             <property name="label" translatable="yes">We won't touch your disks until you click 'Begin Installation'.</property>
+            <property name="xalign">1</property>
             <attributes>
               <attribute name="font-desc" value="Cantarell Italic"/>
               <attribute name="foreground" value="#808080808080"/>
-- 
2.16.2

